src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -438,21 +471,92 @@ public static int CursorTop
             }
         }
 
+        // Although msdn states that the max allowed limit is 65K,
+        // desktop limits this to 24500 as buffer sizes greater than it
+        // throw.
+        private const int MaxConsoleTitleLength = 24500;
+
         public static string Title
         {
-            // TODO #4636: Implement this
-            get { throw new NotImplementedException(); }
-            set { throw new NotImplementedException(); }
+            [System.Security.SecuritySafeCritical]  // auto-generated
```

Comment:
We have tried to be consistent with security attributes in the past and so, I let them stay.